yt/frontends/ramses/data_structures.py

```diff
@@ -260,7 +260,9 @@ def _read_particle_header(self):
                 ("particle_mass", "d"),
                 ("particle_identifier", "i"),
                 ("particle_refinement_level", "I")]
-
+            if hvals["nstar_tot"] > 0:
+                particle_fields += [("particle_formation_time", "d"),
```

Comment:
This means that if we have a time serie of datasets, the `particle_formation_time` will only be created once we have stars in the simulation, which is somehow inconsistent.

Reply:
I don't have access to any RAMSES time series that form stars dynamically. Do you happen to have any? It's not clear to me if the formation time fields are even in the outputs before star particles start to form.

Reply:
The fields are the same for a unique RAMSES run (it depends on the compilation flags + the run parameters only), so the `particle_formation_time` should be present as long as the code *may* create stars, i.e. the parameters controlling star formation aren't 0.
It is possible though to stop a run, tweak the parameters so that star formation may happen. In this case, the fields would appear at the first output after restart.

Reply:
I'd really like to have a dataset that includes these fields but doesn't have any stars yet. Perhaps there's a better heuristic we can use than looking for `nstart_tot`? It's hard to say without any actual datasets to work with. FWIW the `nstar_tot` check is what we were doing in yt 3.4.0, so this wouldn't new behavior.

Reply:
I'm generating a dataset for you.

Reply:
Well, there are two cases. 

1. for users using the new RAMSES output format, the file descriptor tells about a field called `birth_time` (which is the `particle_formation_time`).
2. for users on an older version of RAMSES, if `nstar_tot > 0`, it's indeed easy. But if there's no star, I don't know…